File: client/src/app/shared/models/aluno.ts
import { Telefone } from "./telefone";
import { Endereco } from "./endereco";
import { Matricula } from "./matricula";

export interface Aluno {
    id: number;
    nome: string;
    sexo: string;
    endereco: Endereco;
    enderecoId: number;
    telefoneId: number;
    telefone: Telefone;
    dataNascimento: Date   
}

File: server/edusys.Api/Entities/Aluno.cs
using edusys.Api.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace edusys.Api.Entities
{
    [Table("Aluno")]
    public class Aluno
    {
        [Key]
        public int Id { get; set; }
        public string? Nome { get; set; }
        public string? Sexo { get; set; }
        public DateTime DataNascimento { get; set; }
        public int EnderecoId { get; set; }
        public virtual Endereco Endereco { get; set; }
        public int? TelefoneId { get; set; }
        public virtual Telefone Telefone { get; set; }

    }
}
